### PR TITLE
[CHANGE] Able to bring up color palette to style selected custom crea…

### PR DESCRIPTION
**Pull request by achenPDFTron**
…ted annotations

### CODE REVIEW

File: src/apis/registerTool.js [hunk 1]
```diff
@@ -22,18 +23,18 @@ WebViewer(...)
...
+    instance.registerTool(myTool, undefined, annot => annot && annot.isCustomAnnot);
   });
  */
 
 import core from 'core';
 import { register, copyMapWithDataProperties } from 'constants/map';
 import actions from 'actions';
-
-export default store => (tool, annotationConstructor) => {
+// TODO may want to remove annotationConstructor in 6.4.0 as custoomANnotationCheckFxn makes it sort of redundant
```
Comment (ZhijieZhang) [inline src/apis/registerTool.js:33]: It should be `7.0` instead of `6.4.0`

File: src/apis/registerTool.js [hunk 2]
```diff
@@ -9,6 +9,7 @@
  * @param {string} [properties.buttonGroup] Group of the tool button belongs to.
  * @param {string} [properties.tooltip] Tooltip of the tool button.
  * @param {function} [annotationConstructor] The constructor function for the annotation that will be created by the registered tool.
+ * @param {function} [customAnnotationCheckFxn] Function that takes in a parameter of an annotation. Returns a boolean if the specified annotation is a certain type of annotation. When this is defined, annotationConstructor can be omitted. Must define it when creating a custom annotation
```
Comment (ZhijieZhang) [inline src/apis/registerTool.js:12]: Is `Fxn` a widely used word? Can we use `Func` or `Function` instead?

Also, can we change this description to mention something like, `this function is used by the viewer to check if the annotation passed in is associated(created) with the registered tool`?

File: src/apis/registerTool.js [hunk 3]
```diff
@@ -9,6 +9,7 @@
  * @param {string} [properties.buttonGroup] Group of the tool button belongs to.
  * @param {string} [properties.tooltip] Tooltip of the tool button.
  * @param {function} [annotationConstructor] The constructor function for the annotation that will be created by the registered tool.
```
Comment (ZhijieZhang) [inline src/apis/registerTool.js:11]: Can we change the description to this argument to say it's deprecated, and the third argument should be used? Thanks!

File: src/apis/registerTool.js (hunk 1 above)
Comment (achenPDFTron) [reply src/apis/registerTool.js:33]: addressed

**Pull request closed without merging**